### PR TITLE
Updated Offscreen page

### PR DESCRIPTION
**Pull request by wizardofdocs**
Cleaned up phrasing, removed an unnecessary table. Questions in comments.

### CODE REVIEW

File: site/en/docs/extensions/reference/offscreen/index.md [hunk 1]
```diff
@@ -97,12 +113,14 @@ chrome.action.onClicked.addListener(async () => {
...
-For complete examples, see the [offscreen-clipboard][gh-offscreen-clipboard] and [offscreen-dom][gh-offscreen-dom] demos on GitHub.
+For complete examples, see the [offscreen-clipboard][gh-offscreen-clipboard] and
+[offscreen-dom][gh-offscreen-dom] demos on GitHub.
 
 ### Before Chrome 116: check if an offscreen document is open
 
 [`runtime.getContexts()`][runtime-get-contexts] was added in Chrome 116. In earlier versions of
-Chrome, you can check for the existence of the offscreen document using [`clients.matchAll()`](https://developer.mozilla.org/docs/Web/API/Clients/matchAll):
+Chrome, use [`clients.matchAll()`](https://developer.mozilla.org/docs/Web/API/Clients/matchAll)
+to check for an existing offscreen document:
```
Comment (jpmedley) [inline site/en/docs/extensions/reference/offscreen/index.md:123]: This code has problems. Please replace it with this:

```js
async function hasOffscreenDocument() {
    if (chrome.runtime.getContexts) {
        return chrome.runtime.getContexts();
    else {
        const matchedClients = await clients.matchAll();
        return matchedClients.some(client => {
            client.url.includes(chrome.runtime.id);
        });
    }
}
```

Comment (jpmedley) [inline site/en/docs/extensions/reference/offscreen/index.md:123]:
```suggestion
to check for an existing offscreen document. To cover both Chrome 116 or later, and earlier than 116, you can do something like the code below. Alternatively, you can set the [`"minimum_chrome_version"`](/docs/extensions/mv3/manifest/minimum_chrome_version/) to 116.
```

**Review by jpmedley — CHANGES_REQUESTED** [review]
I had a lot of fact checking to do on this one. Other than answering your questions, none of the requested changes have anything to do with the work that you did. Who knows how long these errors would have lived on the site if you weren't helping us quality check these.

File: site/en/docs/extensions/reference/offscreen/index.md [hunk 2]
```diff
@@ -2,11 +2,16 @@
...
+limit the functionality of content scripts. The Offscreen API allows the extension to use DOM
+APIs in a hidden document without interrupting the user experience by opening new windows or
+tabs. This is the only Chrome API that offscreen documents support. <!--What does DOM stand for?-->
 
 ## Manifest
 
-You must declare the `"offscreen"` permission in the [extension manifest][doc-manifest] to use the Offscreen API. For example:
+You must declare the `"offscreen"` permission in the [extension manifest][doc-manifest] to use
+the Offscreen API. For example: 
+<!--I'd like to make the standard phrasing "To use the Foo API, declare the `foo` permission in the [extension manifest][doc-manifest]. For example:" This would require updating this section in all the docs.-->
```
[the remark this answers is not in the document]
Comment (wizardofdocs) [reply site/en/docs/extensions/reference/offscreen/index.md]: Done.

File: site/en/docs/extensions/reference/offscreen/index.md (hunk 1 above)
Comment (wizardofdocs) [reply site/en/docs/extensions/reference/offscreen/index.md:123]: Does this change still apply to the new code?

Comment (jpmedley) [reply site/en/docs/extensions/reference/offscreen/index.md:123]: Yes. Ask me about feature detection dome time.

File: site/en/docs/extensions/reference/offscreen/index.md [hunk 3]
```diff
@@ -97,23 +113,25 @@ chrome.action.onClicked.addListener(async () => {
...
-      }
+async function hasOffscreenDocument() {
+    if (getContexts in chrome.runtime) {
+        return chrome.runtime.getContexts();
+    } else {
+        const matchedClients = await clients.matchAll();
+        return await matchedClients.some(client => {
+            client.url.includes(chrome.runtime.id);
+        });
     }
```
Comment (jpmedley) [inline site/en/docs/extensions/reference/offscreen/index.md:134]:
```suggestion
  if (chrome.runtime.getContexts) {
    const contexts = await chrome.runtime.getContexts({
      contextTypes: ['OFFSCREEN_DOCUMENT'],
      documentUrls: [OFFSCREEN_DOCUMENT_PATH]
    });
    return Boolean(contexts.length);
  } else {
    const matchedClients = await clients.matchAll();
    return await matchedClients.some(client => {
        client.url.includes(chrome.runtime.id);
    });
  }
```

File: site/en/docs/extensions/reference/offscreen/index.md (hunk 1 above)
Comment (jpmedley) [reply site/en/docs/extensions/reference/offscreen/index.md:123]: Even though Chrome autoupdates, enough users have it blocked that older versions need to be taken into account in web and extension development. An enterprise extension can set a minimum. The developer of a web store extension can set a minimum version but probably shouldn't.

**Pull request merged**